projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-header-row.component.ts

```diff
@@ -275,25 +275,23 @@ export class IgxPivotHeaderRowComponent extends IgxGridHeaderRowComponent {
                 .concat(this.grid.pivotConfiguration.columns)
                 .concat(this.grid.pivotConfiguration.filters);
             // chip moved from external collection
-            const dims = allDims.filter(x => x && x.memberName === dragId);
-            if (dims.length === 0) {
+            const dim = allDims.find(x => x && x.memberName === dragId);
+            if (!dim) {
                 // you have dragged something that is not a dimension
                 return;
             }
-            dims.forEach(element => {
-                element.enabled = false;
-            });
-
+            const dimType = this.getDimensionsType(dim);
             const currentDimChild = currentDim.find(x => x && x.memberName === dragId);
             if (currentDimChild) {
                 currentDimChild.enabled = true;
```

Comment:
Is there a case after your changes where currentDimChild wil not be undefined?

Reply:
Yes. I think that check can be removed.